modules/programs/mu.nix

```diff
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.mu;
+
+  # By default, mu will place it's database in XDG_CACHE_HOME/mu.
+  defaultDBLocation = config.xdg.cacheHome + "/mu";
+
+  # Used to generate command line arguments that mu can operate with.
+  genCmdMaildir = path: "--maildir=" + path;
+
+  genCmdAddrs = addrs:
+    let
+      # Takes the list of address strings and returns list of strings with the
+      # --my-address= prefixed to each
+      addMyAddr = addrs: builtins.map (addr: "--my-address=" + addr) addrs;
+    in builtins.concatStringsSep " " (addMyAddr addrs);
+
+in {
+  meta.maintainers = [ maintainers.KarlJoad ];
+
+  options.programs.mu = {
+    enable = mkEnableOption "mu, a maildir indexer and searcher";
+    # No options/config file present for mu, and program author will not be adding
+    # one soon.
+    # See https://github.com/djcb/mu/issues/882 for more information about this.
+
+    # All configuration options below affect only the initial index script wrapper.
+
+    maildir = mkOption {
+      type = types.path;
+      default = config.accounts.email.maildirBasePath;
+      defaultText = "$HOME/Maildir";
+      example = "$HOME/Mail";
+      description = ''
+        Path to the base directory for all accounts' maildir directories. May be
+        a relative path or absolute path. Must be specified for the initial index,
+        then, never again.
+
+        Note that if you define your own path, you cannot leave a trailing
+        <literal>/</literal>.
+      '';
+    };
+
+    addresses = mkOption {
```

Comment:
I would suggest removing this option and instead add an enable option like for the other email related modules. Otherwise it might be confusing, to have all modules work the same except the mu module. Like in `notmuch.nix`, for example:
https://github.com/nix-community/home-manager/blob/43ab2f40b92030435bc0afe9ecf573be4df03375/modules/programs/notmuch.nix#L147-L152

Reply:
I have changed `mu.nix` to have this as well. Will be committed soon.